src/ocean/text/convert/Formatter.d

```diff
@@ -381,9 +381,22 @@ private void handle (T) (T v, FormatInfo f, scope FormatterSink sf, scope ElemSi
     static if (is(T == typeof(null)))
         se("null", f);
 
-    // Cannot print enum member name in D1, so just print the value
+    // Print enum type and member name
     else static if (is (T V == enum))
-        handle!(V)(v, f, sf, se);
+    {
```

Comment:
The base type of the enum only works normally with an integer or string.
This causes errors in other base types.
``` D
    import std.stdio;
    enum FooB : real {
        one = 1,
        two = 1.41421,
        three = 1.73205
    }
    FooB fb = FooB.one;
    writeln(format("{}", fb));
```

Reply:
For enums where you can't do a `switch` it's possible to do a linear search like this:

```D
void handleNonSwitchableEnum()
{
    static foreach (idx, member; EnumAllMembers!T)
    {
        if (v == member)
            sf(T.stringof ~ "." ~ __traits(identifier, EnumAllMembers!T[idx]));
            return;
    }

    sf("cast(" ~ T.stringof ~ ") ");
    handle!(V)(v, f, sf, se);
}

handleNonSwitchableEnum();
```

But it's not very efficient because of `O(n)` comparisons.

I could think of another solution: to hash the value and reduce it to a `ulong` which can then be used in a switch/case. But this would require a lot more code. Also it's not currently possible with our hashing routines in `Hash.d` because they cannot run at compile-time (CTFE): `source/agora/common/Hash.d(118,28): Error: `crypto_generichash_init` cannot be interpreted at compile time, because it has no available source code`.

Reply:
What is the error you are seeing ?

Reply:
``` D
sw: switch (v)
```
I was thinking about when other types of data came because v is only an integer and string.

Reply:
Right, good point. Honestly non-integral / string enums are so rare that we can leave it out for the time being. Just make it fall back to the underlying type when that happens.

Reply:
That's a good idea.